core/src/test/java/com/netflix/iceberg/TestSplitScanTaskIterator.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.netflix.iceberg;
+
+import com.google.common.collect.Lists;
+import com.netflix.iceberg.expressions.Expression;
+import org.junit.Assert;
+import org.junit.Test;
+import java.util.List;
+import static com.netflix.iceberg.BaseFileScanTask.SplitScanTaskIterator;
+
+public class TestSplitScanTaskIterator {
+  @Test
+  public void testSplits() {
+    long splitSize = 15L;
+    List<FileScanTask> tasks = Lists.newArrayList(new SplitScanTaskIterator(splitSize, new MockFileScanTask(100L)));
+    Assert.assertEquals(7, tasks.size());
+    long offset = 0;
+    for (int i = 0; i < tasks.size(); i++) {
```

Comment:
This should use a hard-coded set of expected splits to validate the results.

That is much easier to read and we don't have to worry about the correctness of the arithmetic in the tests as well as in the split code.

Reply:
That makes a lot of sense @rdblue